docs/design/backups.md

```diff
@@ -0,0 +1,83 @@
+# Backup & Restore
+
+In particularly disastrous situations, we would like to be able to revert the etcd cluster to a previously known healthy state.
+This document lays out the design of how we want the healthy state to be captured, securely stored & safely restored to operator-managed etcd clusters.
+
+## Backup Resource
+
+We will introduce a new custom resource: `EtcdBackup`.
+
+```yaml
+apiVersion: etcd.improbable.io/v1alpha1
+kind: EtcdBackup
+metadata:
+  name: my-cluster-backup
+  namespace: default
```

Comment:
Presumably an EtcdBackup exists in the same namespace as the EtcdCluster it is backing up?

Reply:
i've modified the resource to take a `clusterAddress` to select the cluster to back up - this means the `EtcdCluster` may be in a separate namespace, or might not even be an `EtcdCluster` at all. thoughts?

Reply:
Using a `clusterAddress` and creating something generic for all etcd clusters means we'll need to also include more cluster connection information, such as TLS caBundles, potentially client certs, and maybe also different resolution mechanisms (i.e. SRV lookup vs A record lookup etc). I'm not against that, but it does limit how much 'magic' we can sprinkle in (unless we encapsulate that magic in a _separate_ controller, which also seems fairly reasonable)

Reply:
(Sorry for unresolve, but I had a follow-on question.)

If we're doing connection instructions as below, and the controller is dialling etcd itself. Then does it even make sense for a `EtcdBackup` to have a namespace? Should it be a global resource?

Reply:
possibly that makes sense for now, but as you mentioned in your comment below, we may eventually want the operator to take the backup in another pod, which would need a namespace. unless there's a compelling downside to keeping the resource namespaced i think its safer to leave as is. thoughts?